File: 1614. Maximum Nesting Depth of the Parentheses.java
// Easy
// Iteration
// Time:O(n),Space:O(1)
// https://leetcode.cn/problems/maximum-nesting-depth-of-the-parentheses/

class Solution {
  public int maxDepth(String s) {
      int count = 0, max = 0;
      for (char c : s.toCharArray()) {
          if (c == '(') {
              count++;
              max = count > max ? count : max;
          } else if (c == ')') {
              count--;
          }
      }
      return max;
  }
}

File: 1001-2000/1614. Maximum Nesting Depth of the Parentheses.py
# Time:O(n), Space:O(1)

class Solution:
    def maxDepth(self, s: str) -> int:
        cur_depth, max_depth = 0, 0
        
        for c in s:
            if c == "(":
                cur_depth += 1
                max_depth = max(max_depth, cur_depth)
            elif c == ")":
                cur_depth -= 1
        
        return max_depth
    